src/image_tools/bake.py

```diff
@@ -159,18 +156,71 @@ def bake_command(args: Namespace, product_name: str, bakefile) -> Command:
     )
 
 
+def filter_product_version(conf, product_name: Optional[str], product_version: Optional[str],
+                           product_version_quantile: Optional[int], max_bake_runners: int) -> Optional[int]:
+    """ Filter product versions by the given product_version argument.
+        Mutates the "conf.products" array to remove version of "product_name" that don't match "product_version"
+
+        Returns the number of product versions to build.
+    """
+    result = None
+    if product_name and (
+            product_version or str(product_version_quantile)):  # str() is hack for the case when the quantile is 0
+        filtered_products = []
+        for product in conf.products:
+            if product["name"] == product_name:
+                filtered_product_versions = []
+                for index, version_dict in enumerate(product.get("versions", [])):
+                    if bake_product_version(version_dict["product"], index, product_version, product_version_quantile,
+                                            max_bake_runners):
+                        filtered_product_versions.append(version_dict)
+                # Make a copy of the product and replace the "versions" array with the version that matched.
+                filtered_product = product
+                filtered_product["versions"] = filtered_product_versions
+                filtered_products.append(filtered_product)
+                result = len(filtered_product_versions)
```

Comment:
`result` will be invalid if we match multiple products.

Reply:
Currently we only support building one product at a time.